homeassistant/components/http/__init__.py

```diff
@@ -195,13 +199,28 @@ def __init__(self, hass, development, api_password, ssl_certificate,
         self.app[KEY_LOGIN_THRESHOLD] = login_threshold
         self.app[KEY_DEVELOPMENT] = development
 
+        try:
+            sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            sock.bind((server_host, server_port))
```

Comment:
I couldn't find a definitive answer from the Python docs. It requires syscalls, but there should be no IO required since the port binding is internal to the kernel. I'll roll this back anyway as discussed below.